internal/storage/bucket_handle_test.go

```diff
@@ -467,3 +469,94 @@ func (t *BucketHandleTest) TestUpdateObjectMethodWithMissingObject() {
 
 	AssertTrue(errors.As(err, &notfound))
 }
+
+func (t *BucketHandleTest) TestComposeObjectMethodWithValidObject() {
+	obj, err := t.bucketHandle.StatObject(context.Background(),
```

Comment:
instead you should stat on the destination object and show that object doesn't exist

Reply:
fixed